packages/vite/src/node/plugins/css.ts

```diff
@@ -1545,7 +1545,7 @@ function cleanScssBugUrl(url: string) {
     typeof window !== 'undefined' &&
     typeof location !== 'undefined'
   ) {
-    const prefix = location.href.replace(/\/$/, '')
+    const prefix = location.href?.replace(/\/$/, '')
```

Comment:
I think we could replace 1546 with `typeof location?.href === 'string'`. If `href` isn't defined, then we don't need to call `replace`

Reply:
@mschakulat you haven't fixed this part yet, which I agree would be a nicer check

Reply:
@bluwy Yes, that's correct. I have adjusted it.

Reply:
Unfortunately condition must be `typeof location !== 'undefined'`. 
In my project with svelte-kit and scss doesn't work new condition, because location is not defined.

Optional chaining doesn't work on not defined variables.